### PR TITLE
Push from aleph -> concat

### PR DESCRIPTION
**Pull request by yusefnapora**
quick summary:

- it works!
- it's ugly
- it'll get all nice like after the break :smile:

### CODE REVIEW

**Comment by yusefnapora** [comment]
okay, I think this is in decent shape, although maybe we should merge #146 first, since this branches off of it.  I could also do a bit of squashing to sand off some of the edges in the commit history 😄 

The last commit adds an `ingestSimpleStatement` method to the aleph `MediachainNode` class - it's not as fancy as the mcclient publish in that it doesn't try to do any id extraction, etc.  It just takes a namespace, "body" object, refs (and optionally tags + deps).  Then adds the serialized object to the datastore and makes + signs a statement and adds it to the DB.  It's used in `integration_test/push_test.js` and seems to be working.

The node's publisher id is currently optional, which isn't ideal... it needs to be generated asynchronously, so you can't make one in the constructor.  And I didn't want to make it mandatory since we'll need to update existing usages.  The ingest command will fail with an error if it's not set.

File: integration-test/push_test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      alephNode = new AlephNode({peerId, publisherId})
+    })
+    .then(() => seedStatementsToAleph(alephNode))
+    .then(_statementIds => { statementIds = _statementIds })
+    .then(() => seedUnauthorizedStatement(alephNode))
+    .then(_stmtId => { unauthorizedStatementId = _stmtId })
+    .then(() => concatNodeClient())
+    .then(concat => concat.authorize(alephPeerIdB58, ['scratch.*']))
+    .catch(err => {
+      console.error('error during push setup:', err)
```
Comment (parkan) [inline integration-test/push_test.js]: do we really want to catch this?

File: src/peer/datastore.js [hunk 2]
```diff
@@ -37,7 +38,12 @@ class Datastore {
     }
 
     levelOpts.valueEncoding = valueCodec
-    this.db = Levelup(levelOpts)
+    const location = (options.location == null || options.location === '')
+      ? '/aleph/data-' + uuid.v4()
+      : options.location
```
Comment (parkan) [inline src/peer/datastore.js]: `const location = options.location || '/aleph/data-' + uuid.v4()`

Unless you want `number` 0, `NaN` and `false` to be valid locations?

(there's a moment of wondering if js messes up the precedence in this but it's actually OK)

File: integration-test/push_test.js (hunk 1 above)
Comment (yusefnapora) [inline integration-test/push_test.js]: ah, sorry bout that! I stuck that in when things were failing during setup so I could print a stack trace - need to remove it.  well spotted :)

**Comment by parkan** [comment]
OK is this ready for final review?

**Comment by yusefnapora** [comment]
I think so; although I realized I don't have a test for a partially-successful push (some valid results + error).  I think I'll add that real quick; hopefully it'll work as expected :)

File: integration-test/push_test.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+const seedObjects = [
+  {id: 'foo:1', foo: 'bar'},
+  {id: 'foo:2', foo: 'baz'}
+]
+
+function seedStatementsToAleph (alephNode: AlephNode): Promise<Array<string>> {
+  return Promise.all(
+    seedObjects.map(obj =>
+      alephNode.ingestSimpleStatement('scratch.test', obj, { refs: [obj.id] })
```
Comment (parkan) [inline integration-test/push_test.js]: one comment here would be towards separating the test flow and the normal flow, i.e. not using `ingest*` functions to seed test data, but this is probably a separate discussion around testing methodology

**Pull request merged**